htdocs/core/js/lib_head.js.php

```diff
@@ -529,6 +529,7 @@ function hideMessage(fieldId,message) {
  * @param   string  token       Token
  */
 function setConstant(url, code, input, entity, strict, forcereload, userid, token) {
+	var $url = url; /* avoid undefined url */
```

Comment:
"$url"   ?  
Are you sure. We are in javascript context.

Reply:
@eldy yes, this a global variable for avoid "undefined url" inside "function{}"
```
$.post( $url, {
						action: "set",
						action: "set",
						name: key,
						name: key,
						value: value,
						value: value,
						entity: entity
						entity: entity,
						token: token
					});
					});
```

Reply:
But if the variable is undefined, it means we should not call the function and just make a "return" before.

Reply:
I think the problem with the CSRF error is linked with the renewal of the token by the ajax request. 

So we should add this on the php side: 

if (!defined('NOTOKENRENEWAL')) define('NOTOKENRENEWAL', '1'); 

But I don't know where... sorry.

Reply:
Yes. Every ajax page must contains this.
First line of code.
Before the include of main.

Reply:
@eldy @daraelmin 
sorry but, I did various tests and it is necessary to pass the field "url" to "$url" (global variable) so that the value of "url" is retrieved in the "function" to be used in "$ .post($url, ....."  !!